nestkernel/ring_buffer.cpp

```diff
@@ -33,21 +33,24 @@ nest::RingBuffer::resize()
   size_t size = Scheduler::get_min_delay() + Scheduler::get_max_delay();
   if ( buffer_.size() != size )
   {
-    buffer_.resize( size );
-    buffer_ = 0.0;
+    std::vector< double_t >( size, 0.0 ).swap( buffer_ );
   }
 }
 
 void
 nest::RingBuffer::clear()
 {
-  resize();      // does nothing if size is fine
-  buffer_ = 0.0; // clear all elements
+  resize(); // does nothing if size is fine
+  // clear all elements
+  for ( std::vector< double_t >::iterator it = buffer_.begin(); it != buffer_.end(); ++it )
```

Comment:
Isn't there a copy iterator/operator in the STL that allows one to turn this explicit for-loop into a single line?